index_votes.py

```diff
@@ -39,108 +39,96 @@ def gen_records(cur, query):
 
 def _index_votes(pg_conn, _client, _chain_num):
     with pg_conn.cursor() as cur:
-        all_chain_nums = [
-            record[0] for record in gen_records(cur, "select num from chain;")
-        ]
-        max_block_heights = {
-            chain_num: max_block_height
-            for chain_num, max_block_height in gen_records(
-                cur,
-                "select chain_num, MAX(block_height) from votes group by chain_num;",
+        cur.execute(
+            "SELECT MAX(block_height) FROM votes WHERE chain_num = %s", (_chain_num,)
+        )
+        res = cur.fetchone()
+        max_block_height = 0 if res[0] is None else res[0]
+        logger.debug(f"{_chain_num=} {max_block_height=}")
+        for event in new_events_to_process(
+            cur, "votes", _chain_num, max_block_height
+        ):
```

Comment:
These are the only lines I'm really changing in the PR, the rest are only un-indented from removing the for loop.

This catches the edge case when there are no votes in the table and `MAX(block_height) == null`, so default to 0.